pydropsonde/helper/rawreader.py

```diff
@@ -40,69 +40,61 @@ def check_launch_detect_in_afile(a_file: "str") -> bool:
     bool
         True if launch is detected (1), else False (0)
     """
+    if a_file is not None and os.path.getsize(a_file) > 0:
+        with open(a_file, "r") as f:
+            module_logger.debug(f"Opened File: {a_file=}")
+            lines = f.readlines()
 
-    with open(a_file, "r") as f:
-        module_logger.debug(f"Opened File: {a_file=}")
-        lines = f.readlines()
-
-        for i, line in enumerate(lines):
-            if "Launch Obs Done?" in line:
-                line_id = i
-                module_logger.debug(f'"Launch Obs Done?" found on line {line_id=}')
-                break
-
-        return bool(int(lines[line_id].split("=")[1]))
+            for i, line in enumerate(lines):
+                if "Launch Obs Done?" in line:
+                    line_id = i
+                    module_logger.debug(f'"Launch Obs Done?" found on line {line_id=}')
+                    break
 
+            return bool(int(lines[line_id].split("=")[1]))
+    else:
+        return None
 
-def get_sonde_id(a_file: "str") -> str:
-    """Returns Sonde ID for a given A-file
 
-    Given the path for an A-file, the function parses through the lines
-    till it encounters the phrase 'Sonde ID/Type' and returns the sonde ID as a string.
+def get_sonde_id(d_file: "str") -> str:
+    """Returns Sonde ID for a given D-file
 
-    The function splits the line with the aforementioned phrase at the first ':' sign.
-    It takes the succeeding string and splits it again at the first ','.
-    It then takes the preceding string, removes any whitespace on the left side and returns
-    the string as the sonde ID.
+    Given the path for an D-file, the function reads in the first line (header) and extracts the sonde ID.
 
     Parameters
     ----------
-    a_file : str
-        Path to A-file
+    d_file : str
+        Path to D-file
 
     Returns
     -------
     str
         Sonde ID
     """
     try:
-        with open(a_file, "r") as f:
-            module_logger.debug(f"Opened File: {a_file=}")
-            lines = f.readlines()
-
-            for i, line in enumerate(lines):
-                if "Sonde ID/Type" in line:
-                    module_logger.debug(f'"Sonde ID/Type" found on line {i=}')
-                    break
-
-            return lines[i].split(":")[1].split(",")[0].lstrip()
+        with open(d_file, "r") as f:
+            module_logger.debug(f"Opened File: {d_file=}")
+            header = f.readline()
+            return header.split(" ")[2]
     except UnboundLocalError:
-        afile_base = os.path.basename(a_file)
-        return afile_base.split(".")[0][1:]
+        dfile_base = os.path.basename(d_file)
+        return dfile_base.split(".")[0][1:]
 
 
-def get_sonde_rev(a_file: str) -> Optional[str]:
-    with open(a_file, "r") as f:
-        module_logger.debug(f"Opened File: {a_file=}")
+def get_sonde_rev(a_file: str | None) -> Optional[str]:
+    if a_file is not None and os.path.getsize(a_file) > 0:
```

Comment:
This likely works as intended, but one could think of using a guard clause in order to reduce the indentation depth afterwards. E.g. if the function would start with:

```python
if a_file is None:
    return None
```

The rest of the function could be as before, and with one indentation level less.